services/rds/src/main/java/org/finra/gatekeeper/services/db/connections/LambdaConnection.java

```diff
@@ -0,0 +1,194 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+
+/**
+ * Interface for dealing with AWS Lambda.
+ */
+@Component
+public class LambdaConnection  implements DBConnection{
+    private final Logger logger = LoggerFactory.getLogger(LambdaConnection.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final AwsSessionService awsSessionService;
+    private final GKUserCredentialsProvider gkUserCredentialsProvider;
+    private final String gkUserName;
+    private final String lambdaFunctionName;
+
+    @Autowired
+    public LambdaConnection(AwsSessionService awsSessionService,
+                            GatekeeperProperties gatekeeperProperties,
+                            @Qualifier("credentialsProvider") GKUserCredentialsProvider gkUserCredentialsProvider){
+        this.gkUserCredentialsProvider = gkUserCredentialsProvider;
+        this.gkUserName = gatekeeperProperties.getDb().getGkUser();
+        this.lambdaFunctionName = gatekeeperProperties.getLambda().getFunction();
+        this.awsSessionService = awsSessionService;
+
+    }
+
+    public Map ping(){
+        return invokeLambda("ping", "GET", "{}");
+    }
+
+    private Map invokeLambda(String uri, String method, String body){
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        LambdaPayload lambdaPayload = new LambdaPayload().setBody(body)
+                .setHeaders(headers)
+                .setHttpMethod(method.toUpperCase())
+                .setBase64Encoded(false)
+                .setPath("/api/rds/" + uri);
+        try {
+            InvokeRequest invokeRequest = new InvokeRequest().withFunctionName(lambdaFunctionName).withPayload(OBJECT_MAPPER.writeValueAsString(lambdaPayload));
+            AWSLambda awsLambda = awsSessionService.getAwsLambda();
+
+            InvokeResult invokeResult = awsLambda.invoke(invokeRequest);
+            String invokeResultString = new String( invokeResult.getPayload().array());
+            Map lambdaResult = OBJECT_MAPPER.readValue(invokeResultString, Map.class);
+            logger.info(lambdaResult.toString());
+            String statusCode = lambdaResult.get("statusCode").toString();
+            if(!statusCode.equals("200")){
+                throw new GatekeeperException("Unable to reach lambda, Status Code: " + statusCode);
+            }
+            return lambdaResult;
+        } catch (Exception e) {
+            logger.error(e.toString());
+            return Collections.singletonMap("Error", e);
+        }
+    }
+
+    public boolean grantAccess(RdsGrantAccessQuery rdsGrantAccessQuery) throws Exception{
+        LambdaDTO lambdaDTO = new LambdaDTO().withDbEngine(rdsGrantAccessQuery.getDbEngine()).withLambdaQuery(new LambdaQuery(rdsGrantAccessQuery)).withGatekeeperPassword(gkUserCredentialsProvider.getGatekeeperSecret(rdsGrantAccessQuery));
```

Comment:
For a lambda based Database connection, I think it should be the sole responsibility of the Lambda to pull down the gatekeeper user password. The metadata for the "RdsGrantAccessQuery" object should be passed along to the lambda and used to retrieve the secret over there. (Seems like you're already doing a great deal of that with the "LambdaQuery" object)